operations/ftdc.go

```diff
@@ -0,0 +1,286 @@
+package operations
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/mongodb/ftdc"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/mongo-go-driver/bson"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+func FTDC() cli.Command {
+	return cli.Command{
+		Name:  "ftdc",
+		Usage: "tools for running FTDC parsing and generating tools",
+		Flags: []cli.Flag{},
+		Subcommands: []cli.Command{
+			ftdctojson(),
+			jsontoftdc(),
+			ftdctobson(),
+			bsontoftdc(),
+		},
+	}
+}
+
+func ftdctojson() cli.Command {
+	return cli.Command{
+		Name:  "ftdctojson",
+		Usage: "write FTDC data to a JSON file",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "input",
+				Usage: "write FTDC data from this file",
+			},
+			cli.StringFlag{
+				Name:  "output",
+				Usage: "write FTDC data in JSON format to this file, defaults to stdout",
+			},
+			cli.BoolFlag{
+				Name:  "flattened",
+				Usage: "flatten FTDC data",
+			},
+		},
+		Action: func(c *cli.Context) error {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			ftdcPath := c.String("input")
+			if ftdcPath == "" {
+				return errors.New("input is not specified")
+			}
+			jsonPath := c.String("output")
+
+			ftdcFile, err := os.Open(ftdcPath)
+			if err != nil {
+				return errors.Wrapf(err, "problem opening file '%s'", ftdcPath)
+			}
+			defer ftdcFile.Close()
+
+			var jsonFile *os.File
+			if jsonPath == "" {
+				jsonFile = os.Stdout
+			} else {
+				jsonFile, err = os.Create(jsonPath)
+				if err != nil {
+					return errors.Wrapf(err, "problem opening flie '%s'", jsonPath)
+				}
+				defer jsonFile.Close()
+			}
+
+			var iter ftdc.Iterator
+			if c.Bool("flattened") {
+				iter = ftdc.ReadMetrics(ctx, ftdcFile)
+			} else {
+				iter = ftdc.ReadStructuredMetrics(ctx, ftdcFile)
+			}
+
+			for iter.Next(ctx) {
+				jsonFile.WriteString(iter.Document().ToExtJSON(false))
+				jsonFile.WriteString("\n")
+			}
+
+			return iter.Err()
+		},
+	}
+}
+
+func jsontoftdc() cli.Command {
+	return cli.Command{
+		Name:  "jsontoftdc",
+		Usage: "write FTDC data from a JSON file",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "input",
+				Usage: "write JSON data from this file, defaults to stdin",
+			},
+			cli.StringFlag{
+				Name:  "prefix",
+				Usage: "prefix for FTDC filenames",
+			},
+			cli.IntFlag{
+				Name:  "maxChunkSize",
+				Usage: "maximum chunk size",
```

Comment:
I pushed a change to ftdc to make this setting more clear, match the server, and be attainable. should revender and update to match. The setting is fine.